doc/program_services.md

```diff
@@ -0,0 +1,1179 @@
+
+This calling convention is explained in detail in the [IBM z/OS Version 2 Release 4 MVS Programming: Assembler Services
+Guide (SA23-1368-40)](https://www-01.ibm.com/servers/resourcelink/svc00100.nsf/pages/zOSV2R4sa231368/$file/ieaa600_v2r4.pdf). Refer to Chapter 2 - Linkage conventions.
+
+GR15 by convention may contain a return code upon exit.
+
+## Program parameters
+
+There are two methods of passing parameters to a program.
+
+* At compile time using SYSPARM
+* At run time using PARM
+
+In either case enclose the entire parm to be passed in double quotes.
+
+The double quotes are required to handle commas and spaces that otherwise cause command processors to split the parm.
+
+The double quotes are not required if there are no commas or spaces in the text. 
+
+### Compile time - SYSPARM
+
+If single quotes are included in text, they are passed on to &SYSPARM.
+
+There is a limit of 32767 bytes for the text. 
+
+SYSPARM will be transferred to the macro variable &SYSPARM.
+
+``` dos
+mz390 ... "SYSPARM(HELLO WORLD)"
+```
+Access the text by coding:
+
+``` hlasm
+label    DC    C'&SYSPARM'
+...... will translate to 
+label    DC    C'HELLO WORLD'
+```
+
+### Run time - PARM
+
+PARM can be accessed via GR1 at program entry.
+
+GR1 will contain an address that points to a fullword in storage which points to a storage location with the parm length in a halfword followed by the parm value.
+
+``` dos
+ez390 ... "PARM(HELLO WORLD)"
+```
+
+The fullword pointed to by GR1 points to:
+
+``` hlasm
+         DC   H'11',C'HELLO WORLD'`
+```
+
+If single quotes are included around text in PARM they are removed.
+
+``` dos
+ez390 .... "PARM('HELLO WORLD')" 
+```
+
+Will also result in the fullword pointed to by GR1 pointing to:
+
+``` hlasm
+         DC   H'11',C'HELLO WORLD'`
+```
+
+#### Usage
+
+The program is run with the parameter passed
+
+``` dos
+ez390 myprog "PARM('HELLO THERE')"
+```
+
+The program code:
+
+``` hlasm
+         L     R2,0(,R1)       R2=address of parm length(HW)+value
+         LH    R3,0(,R2)       R3=length of parm value
+         BCTR  R3,0            Subtract 1 from R3 - length for use in machine code 
+         EX    R3,MOVE         Execute MOVE instruction using length code from R3
+......
+MOVE     MVC   PARM(0),2(R2)   Move instruction used by EX
+PARM     DC    CL80' '         Work area for parm value
+```
+* GR1 is populated by ez390 with the parm address pointer.
+* GR2 is loaded with the address of the parm length and address - fullword pointed to by GR1.
+* GR3 is loaded with the parm length - halfword value pointed to by GR2.
+* GR3 would contain decimal 11 which is the length of the parm "HELLO THERE".
+* GR3 is decremented by 1, making it a length code.
+* The actual parm value of "HELLO THERE" begins 2 bytes past the address in GR2.
+* The `EX` instruction executes the instruction at label MOVE which will move the parm to the storage at label PARM using length from R3.
```

Comment:
Line 130. Slight rewording.

"... which will move the ... using length from R3." -> "... which will copy the ... using the length code in R3."

Reply:
Fixed in dca1f6c.